sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -265,6 +267,15 @@ object Literal {
       s"Literal must have a corresponding value to ${dataType.catalogString}, " +
       s"but class ${Utils.getSimpleName(value.getClass)} found.")
   }
+
+  def fromSQL(sql: String): Expression = {
+    CatalystSqlParser.parseExpression(sql).transformUp {
+      case u: UnresolvedFunction =>
+          assert(u.nameParts.length == 1)
+          assert(!u.isDistinct)
```

Comment:
can we check other fields in `UnresolvedFunction` as well? The SQL string produced by `Literal#sql` should not specify any extra fields

Reply:
Done, i added a bunch of asserts.  Not too familiar with these flags, please double check